TACC/Core-Portal
# Task/WP-72:  Highlight matching search terms

## Overview

[PR FP-319](https://github.com/TACC/Core-Portal/pull/756) introduced the job history search feature. This PR achieves highlighting of the matching search terms/ query in the Job History Search.

## Related

* [WP-72](https://jira.tacc.utexas.edu/browse/WP-72)
* [FP-319](https://jira.tacc.utexas.edu/browse/FP-319)

## Changes

* Created a new component `HighlightSearchTerm` that highlights the matching queries
* Updated new props in InfiniteScrollTable component
* Added a new logic in `InfiniteScrollTable.jsx` for rendering the HighlightSearchTerm component

## Testing

1. Go to the History page in the portal
2. Type query in the searchbar and confirm if the queries are highlighted
3. When no matching term is found in the Job Name and/or Output Location, it will only highlight the 'View Details' button since the query is inside the View Details (the query inside the View Details Modal isn't highlighted)

## UI

![image](https://github.com/TACC/Core-Portal/assets/54924215/f37c0d69-4908-4e78-b2a6-8a1c5177e031)
![image](https://github.com/TACC/Core-Portal/assets/54924215/9d3c2948-7e88-4bc3-8866-70c2b9f93431)
![image](https://github.com/TACC/Core-Portal/assets/54924215/1ae392fa-19e9-4c1f-9ff1-7394c303661d)



## Notes
